File: boj/problem2609/Main.java
package problem2609;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {
    public static String solve(int num1, int num2) {
        int maxNum = Math.max(num1, num2);
        int minNum = Math.min(num1, num2);
        int gcd = 0;
        int lcm = 0;

        for (int i = minNum; i > 0; i--) {
            if (num1 % i == 0 && num2 % i == 0) {
                gcd = i;
                break;
            }
        }

        int increment = maxNum;
        while (true) {
            if (maxNum % num1 == 0 && maxNum % num2 == 0) {
                lcm = maxNum;
                break;
            }
            maxNum += increment;
        }

        return Integer.toString(gcd) + "\n" + Integer.toString(lcm);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int num1 = Integer.parseInt(st.nextToken());
        int num2 = Integer.parseInt(st.nextToken());

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        bw.write(solve(num1, num2));
        bw.flush();
    }
}

File: boj/problem2609/main.py
def solve(num1: int, num2: int) -> str:
    min_num = min(num1, num2)
    max_num = max(num1, num2)
    gcd = 0
    lcm = 0

    for i in range(min_num, 0, -1):
        if num1 % i == 0 and num2 % i == 0:
            gcd = i
            break

    increment = max_num
    while True:
        if max_num % num1 == 0 and max_num % num2 == 0:
            lcm = max_num
            break

        max_num += increment

    return "\n".join(map(str, [gcd, lcm]))


def main():
    num1, num2 = map(int, input().split())

    print(solve(num1, num2))


if __name__ == "__main__":
    main()
